tests/suites/host_test.function

```diff
@@ -565,7 +565,11 @@ int execute_tests( int argc , const char ** argv )
 #endif /* __unix__ || __APPLE__ __MACH__ */
 
                 function_id = strtol( params[0], NULL, 10 );
-                if ( (ret = check_test( function_id )) == DISPATCH_TEST_SUCCESS )
+                if( function_id < 0 )
```

Comment:
We never want `function_id` to be negative. So it should be an unsigned type (`size_t` since it's an array index) and we should parse it with `strtoul`. `check_test` already takes care of checking that's in range, as long as it's non-negative.

Reply:
I have no issues with changing this. As I explained earlier, I only chose this approach so that it catches potential negative input in a slightly more robust way. However, neither it is likely to have a negative input that will pass the upper bounds check, nor it is likely to have an incorrect input in the first place. Therefore I have changed the solution.